smith.lic

```diff
@@ -12,47 +12,75 @@ class Smith
   include DRCT
 
   def initialize(args)
+    ensure_copper_on_hand(5000)
+
     material, item_name, *options = args
+    settings = get_settings(['crafting'])
+    discipline = settings['Blacksmithing']
+    recipes = settings.crafting_recipes.select { |recipe| recipe['type'] =~ /blacksmithing|weaponsmithing|armorsmithing/i }
+    recipe = recipe_lookup(recipes, item_name)
 
-    @settings = get_settings(['crafting'])
+    parts = recipe['part'] || []
+    temper = options.find { |x| x =~ /temper/i }
```

Comment:
I love not needing to do the `|x| x=~` crap, but I like the readability of options.find more. All we really care about is existence here so I'd rather use `.any?` instead of first. I'd be much happier with `options.grep(/temper/i).any?`